nibe/connection/nibegw.py

```diff
@@ -268,19 +266,62 @@ async def read_coil(self, coil: Coil, timeout: float = DEFAULT_TIMEOUT) -> Coil:
             logger.debug(f"Waiting for read response for {coil.name}")
 
             try:
-                await asyncio.wait_for(self._futures["read"], timeout)
+                return await asyncio.wait_for(future, timeout)
             except asyncio.TimeoutError:
                 raise CoilReadTimeoutException(
                     f"Timeout waiting for read response for {coil.name}"
                 )
-            finally:
-                del self._futures["read"]
+            except DecodeException as e:
+                raise CoilReadException(
+                    f"Failed decoding response for {coil.name}: {e}"
+                ) from e
```

Comment:
@elupus If I start capturing exceptions here then it takes `timeout` * 3 to return from the method. Maybe you know how to fix... Couple tests capture that.

Reply:
Ahh, we have that magic retry logic there.... Damn...